### PR TITLE
Only check for leading/trailing space for files on Windows.

### PR DESCRIPTION
**Pull request by camilasan**
Fix for #6807.

### CODE REVIEW

**Review by mgallien — CHANGES_REQUESTED** [review]
can you see how automated tests could be modified to validate your changes ?

**Review by claucambra — APPROVED** [review]

File: test/testsyncvirtualfiles.cpp [hunk 1]
```diff
@@ -835,12 +835,21 @@ private slots:
...
+            QCOMPARE(completeSpy.findItem(fileWithSpaces5)->_status, SyncFileItem::Status::FileNameInvalid);
+            QCOMPARE(completeSpy.findItem(fileWithSpaces6)->_status, SyncFileItem::Status::FileNameInvalid);
+        } else {
+            QCOMPARE(completeSpy.findItem(fileWithSpaces1)->_status, SyncFileItem::Status::Success);
+            QCOMPARE(completeSpy.findItem(fileWithSpaces2)->_status, SyncFileItem::Status::Success);
+            QCOMPARE(completeSpy.findItem(fileWithSpaces3)->_status, SyncFileItem::Status::Success);
+            QCOMPARE(completeSpy.findItem(fileWithSpaces4)->_status, SyncFileItem::Status::Success);
+            QCOMPARE(completeSpy.findItem(fileWithSpaces5)->_status, SyncFileItem::Status::Success);
+            QCOMPARE(completeSpy.findItem(fileWithSpaces6)->_status, SyncFileItem::Status::Success);
+        }
```
Comment (mgallien) [inline test/testsyncvirtualfiles.cpp:852]:
```suggestion
#if defined Q_OS_WINDOWS
            QCOMPARE(completeSpy.findItem(fileWithSpaces1)->_status, SyncFileItem::Status::FileNameInvalid);
            QCOMPARE(completeSpy.findItem(fileWithSpaces2)->_status, SyncFileItem::Status::FileNameInvalid);
            QCOMPARE(completeSpy.findItem(fileWithSpaces3)->_status, SyncFileItem::Status::FileNameInvalid);
            QCOMPARE(completeSpy.findItem(fileWithSpaces4)->_status, SyncFileItem::Status::FileNameInvalid);
            QCOMPARE(completeSpy.findItem(fileWithSpaces5)->_status, SyncFileItem::Status::FileNameInvalid);
            QCOMPARE(completeSpy.findItem(fileWithSpaces6)->_status, SyncFileItem::Status::FileNameInvalid);
#else
            QCOMPARE(completeSpy.findItem(fileWithSpaces1)->_status, SyncFileItem::Status::Success);
            QCOMPARE(completeSpy.findItem(fileWithSpaces2)->_status, SyncFileItem::Status::Success);
            QCOMPARE(completeSpy.findItem(fileWithSpaces3)->_status, SyncFileItem::Status::Success);
            QCOMPARE(completeSpy.findItem(fileWithSpaces4)->_status, SyncFileItem::Status::Success);
            QCOMPARE(completeSpy.findItem(fileWithSpaces5)->_status, SyncFileItem::Status::Success);
            QCOMPARE(completeSpy.findItem(fileWithSpaces6)->_status, SyncFileItem::Status::Success);
#endif
```

File: test/testsyncvirtualfiles.cpp [hunk 2]
```diff
@@ -853,12 +862,21 @@ private slots:
...
+            QCOMPARE(completeSpy.findItem(fileWithSpaces5)->_status, SyncFileItem::Status::Success);
+            QCOMPARE(completeSpy.findItem(fileWithSpaces6)->_status, SyncFileItem::Status::Success);
+        } else {
+            QCOMPARE(completeSpy.findItem(fileWithSpaces1)->_status, SyncFileItem::Status::NoStatus);
+            QCOMPARE(completeSpy.findItem(fileWithSpaces2)->_status, SyncFileItem::Status::NoStatus);
+            QCOMPARE(completeSpy.findItem(fileWithSpaces3)->_status, SyncFileItem::Status::NoStatus);
+            QCOMPARE(completeSpy.findItem(fileWithSpaces4)->_status, SyncFileItem::Status::NoStatus);
+            QCOMPARE(completeSpy.findItem(fileWithSpaces5)->_status, SyncFileItem::Status::NoStatus);
+            QCOMPARE(completeSpy.findItem(fileWithSpaces6)->_status, SyncFileItem::Status::NoStatus);
+        }
```
Comment (mgallien) [inline test/testsyncvirtualfiles.cpp:879]:
```suggestion
#if defined Q_OS_WINDOWS
            QCOMPARE(completeSpy.findItem(fileWithSpaces1)->_status, SyncFileItem::Status::Success);
            QCOMPARE(completeSpy.findItem(fileWithSpaces2)->_status, SyncFileItem::Status::Success);
            QCOMPARE(completeSpy.findItem(fileWithSpaces3)->_status, SyncFileItem::Status::Success);
            QCOMPARE(completeSpy.findItem(fileWithSpaces4)->_status, SyncFileItem::Status::Success);
            QCOMPARE(completeSpy.findItem(fileWithSpaces5)->_status, SyncFileItem::Status::Success);
            QCOMPARE(completeSpy.findItem(fileWithSpaces6)->_status, SyncFileItem::Status::Success);
#else
            QCOMPARE(completeSpy.findItem(fileWithSpaces1)->_status, SyncFileItem::Status::NoStatus);
            QCOMPARE(completeSpy.findItem(fileWithSpaces2)->_status, SyncFileItem::Status::NoStatus);
            QCOMPARE(completeSpy.findItem(fileWithSpaces3)->_status, SyncFileItem::Status::NoStatus);
            QCOMPARE(completeSpy.findItem(fileWithSpaces4)->_status, SyncFileItem::Status::NoStatus);
            QCOMPARE(completeSpy.findItem(fileWithSpaces5)->_status, SyncFileItem::Status::NoStatus);
            QCOMPARE(completeSpy.findItem(fileWithSpaces6)->_status, SyncFileItem::Status::NoStatus);
#endif
```

File: test/testlocaldiscovery.cpp [hunk 3]
```diff
@@ -481,7 +526,15 @@ private slots:
...
         QVERIFY(fakeFolder.currentRemoteState().find(fileTrimmed));
-        QVERIFY(!fakeFolder.currentRemoteState().find(fileWithSpaces));
+
+        if (Utility::isWindows()) {
+            // no file with spaces on Windows
+            QVERIFY(!fakeFolder.currentRemoteState().find(fileWithSpaces));
+        } else {
+            //Linux/Mac OS allows spaces
+            QVERIFY(fakeFolder.currentRemoteState().find(fileWithSpaces));
+        }
```
Comment (mgallien) [inline test/testlocaldiscovery.cpp:536]:
```suggestion
#if defined Q_OS_WINDOWS
            // no file with spaces on Windows
            QVERIFY(!fakeFolder.currentRemoteState().find(fileWithSpaces));
#else
            //Linux/Mac OS allows spaces
            QVERIFY(fakeFolder.currentRemoteState().find(fileWithSpaces));
#endif
```

File: test/testlocaldiscovery.cpp [hunk 4]
```diff
@@ -422,6 +430,43 @@ private slots:
...
+            QCOMPARE(completeSpy.findItem(fileWithSpaces2)->_status, SyncFileItem::Status::Success);
+            QCOMPARE(completeSpy.findItem(fileWithSpaces3)->_status, SyncFileItem::Status::Success);
+            QCOMPARE(completeSpy.findItem(folderWithSpaces1)->_status, SyncFileItem::Status::Success);
+            QCOMPARE(completeSpy.findItem(folderWithSpaces2)->_status, SyncFileItem::Status::Success);
+            QVERIFY(fakeFolder.remoteModifier().find(fileWithSpaces1));
+            QVERIFY(fakeFolder.remoteModifier().find(fileWithSpaces2));
+            QVERIFY(fakeFolder.remoteModifier().find(fileWithSpaces3));
+            QVERIFY(fakeFolder.remoteModifier().find(folderWithSpaces1));
+            QVERIFY(fakeFolder.remoteModifier().find(folderWithSpaces2));
+        }
```
Comment (mgallien) [inline test/testlocaldiscovery.cpp:467]:
```suggestion
#if !defined Q_OS_WINDOWS
            QCOMPARE(completeSpy.findItem(fileWithSpaces1)->_status, SyncFileItem::Status::Success);
            QCOMPARE(completeSpy.findItem(fileWithSpaces2)->_status, SyncFileItem::Status::Success);
            QCOMPARE(completeSpy.findItem(fileWithSpaces3)->_status, SyncFileItem::Status::Success);
            QCOMPARE(completeSpy.findItem(folderWithSpaces1)->_status, SyncFileItem::Status::Success);
            QCOMPARE(completeSpy.findItem(folderWithSpaces2)->_status, SyncFileItem::Status::Success);
            QVERIFY(fakeFolder.remoteModifier().find(fileWithSpaces1));
            QVERIFY(fakeFolder.remoteModifier().find(fileWithSpaces2));
            QVERIFY(fakeFolder.remoteModifier().find(fileWithSpaces3));
            QVERIFY(fakeFolder.remoteModifier().find(folderWithSpaces1));
            QVERIFY(fakeFolder.remoteModifier().find(folderWithSpaces2));
#endif
```

File: test/testlocaldiscovery.cpp [hunk 5]
```diff
@@ -379,16 +389,14 @@ private slots:
         fakeFolder.syncEngine().setLocalDiscoveryOptions(LocalDiscoveryStyle::DatabaseAndFilesystem, {QStringLiteral("foo"), QStringLiteral("bar"), QStringLiteral("bla"), QStringLiteral("A/foo"), QStringLiteral("A/bar"), QStringLiteral("A/bla")});
         QVERIFY(fakeFolder.syncOnce());
 
+#ifdef Q_OS_WINDOWS
```
Comment (mgallien) [inline test/testlocaldiscovery.cpp:392]:
```suggestion
#if defined Q_OS_WINDOWS
```

File: test/testlocaldiscovery.cpp [hunk 6]
```diff
@@ -358,13 +358,23 @@ private slots:
 
         QVERIFY(fakeFolder.syncOnce());
 
+#ifdef Q_OS_WINDOWS
```
Comment (mgallien) [inline test/testlocaldiscovery.cpp:361]:
```suggestion
#if defined Q_OS_WINDOWS
```

**Review by mgallien — CHANGES_REQUESTED** [review]
see my comments

**Review by mgallien — APPROVED** [review]

**Pull request merged**